mevislab.github.io/content/tutorials/thirdparty/pytorch/pytorchexample2.md

```diff
@@ -0,0 +1,177 @@
+
+You should also already add a Python file in the *Commands* section.
+
+{{< highlight filename="DemoAI.script" >}}
+```Stan
+Interface {
+  Inputs {
+    Field inputImage { type = Image }
+  }
+  Outputs {
+    Field outImage { internalName = PythonImage.output0 }
+  }
+  Parameters {
+    Field start { type = Trigger }
+  }
+}
+
+Commands {
+  source = $(LOCAL)/DemoAI.py
+}
+```
+{{</highlight>}}
+
+In MATE, right-click {{< mousebutton "right" >}} the Project Workspace and add a new file *DemoAI.py* to your project. The workspace now contains an empty Python file.
+
+![Project Workspace](/images/tutorials/thirdparty/pytorch_example2_5.png "Project Workspace").
+
+Change to MeVisLab IDE, right-click {{< mousebutton "right" >}} the local macro and select {{< menuitem "Reload Definition">}}. Your new input and output interfaces are now available and you can connect images to your module.
+
+![DemoAI local macro with interfaces](/images/tutorials/thirdparty/pytorch_example2_6.png "DemoAI local macro with interfaces").
+
+### Extend your network
+
+We want to show the segmentation results as an overlay on the original image. Add a `SoView2DOverlayMPR` module and connect it to your `DemoAI` macro. Connect the output of the `SoView2DOverlayMPR` to a `SoGroup`. We also need a lookup table for the colors to be used for the overlay. We already prepared a *\*.xml* file you can simply use. Download the [lut.xml](/examples/thirdparty/pytorch2/lut.xml) file and save it in your current working directory of the project. 
+
+Add a `LoadBase` module and connect it to a `SoMLLUT` module. The `SoMLLUT` needs to be connected to the `SoGroup` so that it is applied to our segmentation results.
+
+![Final network](/images/tutorials/thirdparty/pytorch_example2_7.png "Final network").
+
+{{<alert class="info" caption="Info">}}
+If you have less than 16GB RAM, you should add a `SubImage` module between the `OrthoSwapFlip` and the `Resample3D` module. You should configure less slices in z-direction to prevent your system from running out of memory.
+
+![SubImage module](/images/tutorials/thirdparty/pytorch_example2_7b.png "SubImage module").
+{{</alert>}}
+
+Inspect the output of the `LoadBase` module in the Output Inspector to see if the lookup table has been loaded correctly.
+
+![LUT in LoadBase](/images/tutorials/thirdparty/pytorch_example2_8.png "LUT in LoadBase").
+
+### Write Python script
+You now have to execute the pre-trained PyTorch network on your image. Right-click {{< mousebutton "right" >}} the local macro and select {{< menuitem "Related Files" "DemoAI.script">}}. We want to call a Python function whenever the *Trigger* of your module has been touched.
```

Comment:
You can now execute....

The Python function is supposed to be called whenever the *Trigger* is touched. 
We want the Python function to be executed whenever the *Trigger* is touched.